tests/framework/crossplane.go

```diff
@@ -75,6 +79,19 @@ func ValidateNginxFieldExists(conf *Payload, expFieldCfg ExpectedNginxField) err
 	return fmt.Errorf("field not found; expected: %+v\nNGINX conf: %s", expFieldCfg, string(b))
 }
 
+func validateUpstreamDirectives(expFieldCfg ExpectedNginxField, directive *Directive) error {
+	for _, upstreamName := range expFieldCfg.Upstreams {
+		if directive.Directive == "upstream" && directive.Args[0] == upstreamName {
+			for _, upstreamDirective := range directive.Block {
+				if expFieldCfg.fieldFound(upstreamDirective) {
```

Comment:
I'm not sure if this logic is correct. If there are multiple upstreams in `expFieldCfg.Upstreams`), this will return `nil` as soon as the directive is found in any upstream. Aren't we trying to assert that the directive is present in all upstreams in `expFieldCfg.Upstreams`?

Reply:
I apologize. I thought the upstream would should up for routes we attach it too? Like if we have a route attached to coffee service and that is associated with a usptreamSetingsPolicy? 
Wouldn't it only show up in that coffee upstream block?

Reply:
Yes, that's true. However, in the expFieldCfg, it is possible to specify multiple upstreams:

```
type ExpectedNginxField struct {
...
// Upstream are the upstream names that the directive should exist in.
Upstreams []string
...
}
```

In that case, I would expect this functionality to check that every upstream in expFieldCfg contain the expected directive. 

To be fair, I think this is also an issue with the existing `validateServerBlockDirectives` logic. cc'ing @sjberman in case I'm missing something here